src/main/java/uk/gov/hmcts/reform/et/syaapi/service/VerifyTokenService.java

```diff
@@ -0,0 +1,80 @@
+package uk.gov.hmcts.reform.et.syaapi.service;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.crypto.factories.DefaultJWSVerifierFactory;
+import com.nimbusds.jose.jwk.AsymmetricJWK;
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.JWKSet;
+import com.nimbusds.jose.jwk.SecretJWK;
+import com.nimbusds.jose.proc.JWSVerifierFactory;
+import com.nimbusds.jwt.SignedJWT;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.authorisation.exceptions.InvalidTokenException;
+
+import java.net.URL;
+import java.security.Key;
+
+@Slf4j
+@Service
+public class VerifyTokenService {
+
+    private final JWSVerifierFactory jwsVerifierFactory;
+
+    @Value("${idam.api.jwksUrl}")
```

Comment:
Better to use in a parameter within the constructor so this class is easily mock'able